### DISCUSSION TITLE
How to put object with transfer acceleration on?

### DISCUSSION THREAD

**Discussion by Jasperav** [awslabs/aws-sdk-rust#1006] [Q&A] [answered]
I got this code, but this is using the default bucket. I want to enable transfer acceleration on this upload, how can I do that? 

```
let upload_result = self
    .client
    .put_object()
    .bucket(&*common::properties::AWS_S3_BUCKET_NAME)
    // I am guessing this saves some money
    .storage_class(StorageClass::IntelligentTiering)
    .key(self.path)
    .body(ByteStream::from(self.content))
    .send()
    .await;
```

I also want to check if it's possible to create a presigned URL which use transfer acceleration so that clients can benefit from the performance.

**Comment by rcoh** [comment, marked as answer]
Answered here: https://github.com/awslabs/aws-sdk-rust/issues/1009
